nservicebus/messaging/headers.md

```diff
@@ -73,21 +73,27 @@ A [unique ID for the current message](/nservicebus/messaging/message-identity.md
 
 NServiceBus implements the [Correlation Identifier](http://www.enterpriseintegrationpatterns.com/patterns/messaging/CorrelationIdentifier.html) pattern by using a `Correlation Id` header.
 
-Message correlation connects request messages with their corresponding response messages. The `Correlation Id` of the response message is a `Message Id` of its corresponding request message.
+Message correlation connects request messages with their corresponding response messages. The `Correlation Id` of the response message is the `Correlation Id` of its corresponding request message. Each outgoing message which is sent outside of a message handler will have it's `Correlation Id` set to it's `Message Id`.
```

Comment:
That ends up being true, but it's not how it's written. The code that attaches correlation id is [here](https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Correlation/AttachCorrelationIdBehavior.cs). In pseudocode it is:

- if there is an explicit Correlation Id set, use that
- if there is an incoming message
  - if it has a correlation id header, use that
  - otherwise, if it has a message id header, use that
- if we still don't have a correlation id, use the message id

That means, for the outgoing message without a corresponding incoming message (or an explicitly set correlation id), it gets set to the `Message Id`.  When processing that message, all correlated outgoing messages will get that `Correlation Id` which is the original message's `Message Id`.

So if you say:

1. **Endpoint A** sends **Request** to **Endpoint B**
2. **Endpoint B** handles **Request** and sends **Request 2** to **Endpoint C**

Then the `Correlation Id` on **Request 2** will be the `Message Id` of **Request** even though this isn't a request/response situation. In fact, if you did:

1. **Endpoint A** sends **Request** to **Endpoint B**
2. **Endpoint B** handles **Request** and sends **Response** back to **Endpoint A**
3. **Endpoint A** handles **Response** and publishes **Event**

The the `Correlation Id` on **Request**, **Response**, _and_ **Event** will all be the `Message Id` of **Request**.

`Correlation Id` leaks out of single request/response interactions and works very similarly to `Conversation Id` (which is why this PR was raised in the first place). If that behavior is incorrect then we might need to treat it as a bug and fix it.